src/lucky/remote_ip_handler.cr

```diff
@@ -10,17 +10,26 @@
 class Lucky::RemoteIpHandler
   include HTTP::Handler
 
+  Habitat.create do
+    setting ip_header_names : Array(String) = ["X_FORWARDED_FOR"]
+  end
+
   def call(context)
     context.request.remote_address = fetch_remote_ip(context)
+    context.request.remote_ip = context.request.remote_address.as?(Socket::IPAddress).try(&.address.presence)
     call_next(context)
   end
 
   private def fetch_remote_ip(context : HTTP::Server::Context) : Socket::Address?
     request = context.request
 
-    if x_forwarded = request.headers["X_FORWARDED_FOR"]?.try(&.split(',').first?).presence
+    remote_ip = settings.ip_header_names.compact_map { |header|
+      request.headers[header]?.try(&.split(',').first?).presence
+    }.first?
```

Comment:
I see where you're going with the list of header names, but in practice this will never need to be a list for any one deployment. Here you're iterating and running the entire map on every request.

Is there a way to move this decision so it can be run on demand rather than on every request?

Or perhaps the logic of determining what header might actually contain the value could be done just once?